src/CMakeLists.txt

```diff
@@ -399,7 +491,9 @@ if(CMAKE_CROSSCOMPILING)
   foreach(bootloader ${BOOTLOADERS})
     set(elf ${bootloader}.elf)
     add_executable(${elf} ${BOOTLOADER-SOURCES})
-    target_link_libraries(${elf} PRIVATE c asf4-drivers-min -Wl,-u,exception_table)
+    add_dependencies(${elf} libwally-core)
+    add_dependencies(${elf} noise-c)
+    target_link_libraries(${elf} PRIVATE c asf4-drivers-min noiseprotocol -Wl,-u,exception_table)
```

Comment:
I think noise and libwally shouldn't be needed in the bootloader.

Reply:
Fixed. There was a missing dependency between `libwally.a` and `libwally-core` (the target actually building libwally!), so the dep tree was broken.